### PR TITLE
Add xten_nn.kernel op

### PR DESCRIPTION
**Pull request by mgehre-amd**
fyi @ljfitz

### CODE REVIEW

File: tools/aten-opt/aten-opt.cpp [hunk 1]
```diff
@@ -52,9 +52,7 @@ int main(int argc, char **argv) {
   DialectRegistry registry;
   registerAllDialects(registry);
   mlir::registerAllDialects(registry);
-  registry.insert<amd::xten_nn::XTenNNDialect,
-                  torch::Torch::TorchDialect,
-                  torch::TorchConversion::TorchConversionDialect>();
+  registry.insert<amd::xten_nn::XTenNNDialect>();
```
Comment (TinaAMD) [inline tools/aten-opt/aten-opt.cpp]: This seems unrelated?

File: test/Dialect/XTenNN/ops_invalid.mlir [hunk 2]
```diff
@@ -24,3 +24,32 @@ func.func @mish_int(%arg0: tensor<1x10xi4>) -> tensor<1x10xi4> {
...
+
+func.func @kernel_missing_parenthesis() {
+    // expected-error@+1 {{expected '('}}
+    %a = xten_nn.kernel "myKernel" -> tensor<2xi64>
+}
+
+// -----
+
+func.func @kernel_missing_type(%arg0: i8, %arg1: i8) {
+    // expected-error@+1 {{expected ':'}}
```
Comment (TinaAMD) [inline test/Dialect/XTenNN/ops_invalid.mlir]: Is there a way to specify nicer errors? I.e. it seems we could hint here something like "kernel argument format is val : type, can't find :"

File: test/Dialect/XTenNN/ops_invalid.mlir [hunk 3]
```diff
@@ -24,3 +24,32 @@ func.func @mish_int(%arg0: tensor<1x10xi4>) -> tensor<1x10xi4> {
...
+    %a = xten_nn.kernel "myKernel" (%arg0, %arg1) -> tensor<2xi64>
+}
+
```
Comment (TinaAMD) [inline test/Dialect/XTenNN/ops_invalid.mlir:41]: Can you add a test for a missing type, e.g. ` %a = xten_nn.kernel "myKernel" (%arg0: f32, %arg1:)`?

File: test/Dialect/XTenNN/ops_invalid.mlir [hunk 4]
```diff
@@ -24,3 +24,32 @@ func.func @mish_int(%arg0: tensor<1x10xi4>) -> tensor<1x10xi4> {
...
+func.func @kernel_trailing_comma(%arg0: i8) {
+    // expected-error@+1 {{expected SSA operand}}
+    %a = xten_nn.kernel "myKernel" (%arg0 :i8, ) -> tensor<2xi64>
+}
+
+// -----
+
+func.func @kernel_missing_name() {
+    // expected-error@+1 {{custom op 'xten_nn.kernel' invalid kind of attribute specified}}
```
Comment (TinaAMD) [inline test/Dialect/XTenNN/ops_invalid.mlir]: Error message here could also be more meaningful, we know we are missing the `name` attribute.

File: tools/aten-opt/aten-opt.cpp (hunk 1 above)
Comment (mgehre-amd) [reply tools/aten-opt/aten-opt.cpp]: yes, this belongs in Jose's PR

File: test/Dialect/XTenNN/ops_invalid.mlir (hunk 4 above)
Comment (mgehre-amd) [reply test/Dialect/XTenNN/ops_invalid.mlir]: Unfortunately, having better errors here seems to require to not use any of the parser convenience methods and reimplement most of them. I don't think its worth the effort.

File: test/Dialect/XTenNN/ops_invalid.mlir (hunk 3 above)
Comment (mgehre-amd) [reply test/Dialect/XTenNN/ops_invalid.mlir:41]: Added, but the error message isn't great. Changing it would mean not using the convenience methods of the parser.

File: test/Dialect/XTenNN/ops_invalid.mlir (hunk 2 above)
Comment (mgehre-amd) [reply test/Dialect/XTenNN/ops_invalid.mlir]: this turned out to be possible, so done

**Review by josel-amd — APPROVED** [review]

**Review by TinaAMD — APPROVED** [review]
LGTM
